doc/experiment.rst

```diff
@@ -1,326 +1,350 @@
-
-A simple example of using the Experiment API to create a model and run it
-locally:
-
-.. code-block:: Python
-
-  from smartsim import Experiment
-
-  exp = Experiment("simple", launcher="local")
-
-  settings = exp.create_run_settings("echo", exe_args="Hello World")
-  model = exp.create_model("hello_world", settings)
-
-  exp.start(model, block=True)
-  print(exp.get_status(model))
-
-If the launcher has been specified, or auto-detected through setting
-``launcher=auto`` in the Experiment initialization, the ``create_run_settings``
-method will automatically create the appropriate ``RunSettings`` object and
-return it.
-
-For example with Slurm
-
-.. code-block:: Python
-
-  from smartsim import Experiment
-
-  exp = Experiment("hello_world_exp", launcher="slurm")
-  srun = exp.create_run_settings(exe="echo", exe_args="Hello World!")
-
-  # helper methods for configuring run settings are available in
-  # each of the implementations of RunSettings
-  srun.set_nodes(1)
-  srun.set_tasks(32)
-
-  model = exp.create_model("hello_world", srun)
-  exp.start(model, block=True, summary=True)
-
-  print(exp.get_status(model))
-
-The above will run ``srun -n 32 -N 1 echo Hello World!``, monitor its
-execution, and inform the user when it is completed. This driver script can be
-executed in an interactive allocation, or placed into a batch script as follows:
-
-.. code-block:: bash
-
-    #!/bin/bash
-    #SBATCH --exclusive
-    #SBATCH --nodes=1
-    #SBATCH --ntasks-per-node=32
-    #SBATCH --time=00:10:00
-
-    python /path/to/script.py
-
+:ref:`Model(s)<model_object_doc>` represent a simulation model or any computational kernel,
+including applications, scripts, or generally, a program. They can
+interact with other SmartSim entities via data transmitted to/from
+SmartSim ``Orchestrators`` using a SmartRedis ``Client``.
+
+A ``Model`` is created through the factory method: ``Experiment.create_model()``.
+``Models`` are initialized with ``RunSettings`` objects that specify
+how a ``Model`` should be launched via a workload manager
+(e.g., Slurm) and the compute resources required.
+Optionally, the user may also specify a ``BatchSettings`` object if
+the model should be launched as a batch job on the WLM system.
+The ``create_model()`` factory method returns an initialized ``Model`` object that
+gives you access to functions associated with the :ref:`Model API<model_api>`.
+
+A ``Model`` supports key features, including methods to:
+
+- :ref:`Attach configuration files<files_doc>` for use at ``Model`` runtime.
+- :ref:`Colocate an Orchestrator<colo_model_doc>` to a SmartSim ``Model``.
+- :ref:`Attach a ML model<ai_model_doc>`  into the ``Orchestrator`` at ``Model`` runtime.
+- :ref:`Attach a TorchScript function<TS_doc>`  into the ``Orchestrator`` at ``Model`` runtime.
+- :ref:`Enable SmartSim Model data collision prevention<model_key_collision>`, which allows for reuse of key names.
```

Comment:
"which allows for reuse of key names" -> "which simplifies execution of the same ``Model`` application code"

Reply:
I think that this might be the story for Ensemble to reuse application code that ur suggesting, whereas with Models, keys like input and output might be reused throughout the models that have different application code? Is this correct?